gr-blocks/python/blocks/variable_save_restore.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2023 Philipp Niedermayer.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+
+import numpy
+from gnuradio import gr
+import pickle
+import pmt
+import os
+
+
+class variable_save_restore(gr.sync_block):
+    """
+    (see description in this blocks yml file)
+    """
+
+    def __init__(self, top_block=None, slot="default", variables="", save_trigger=False, restore_trigger=False, description_variable=None):
+        gr.sync_block.__init__(self, name="variable_save_restore", in_sig=[], out_sig=None)
+        self.message_port_register_in(pmt.intern("save"))
+        self.set_msg_handler(pmt.intern("save"), self.handle_msg_save)
+        self.message_port_register_in(pmt.intern("restore"))
+        self.set_msg_handler(pmt.intern("restore"), self.handle_msg_restore)
+
+        self.filename = f"~/.config/gnuradio/{top_block.__class__.__name__}.variables.sav"
```

Comment:
We now have `gr.paths.userconf()`, which returns `.../.config/gnuradio` or a different path if specified by `GR_PREFS_PATH` or derived from `XDG_CONFIG_HOME`.

Reply:
Thanks, changed